File: shared/src/main/java/com/housemate/shared/dto/expense/request/TransactionFilterRequestDTO.java
package com.housemate.shared.dto.expense.request;

import java.util.UUID;

import com.housemate.shared.enums.UserTransactionRole;
import com.housemate.shared.utils.types.DateRange;

public record TransactionFilterRequestDTO(
    UUID householdId,
    UserTransactionRole userTransactionRole,
    DateRange dateRange,
    String description
) {}

File: mobile/lib/shared/dto/expense/request/transaction_filter_request_dto.dart
import 'package:json_annotation/json_annotation.dart';
import '../../../enums/user_transaction_role.dart';
import '../../../utils/types/date_range.dart';

part 'transaction_filter_request_dto.g.dart';

@JsonSerializable(explicitToJson: true)
class TransactionFilterRequestDTO {
  final String? householdId;
  final UserTransactionRole? userTransactionRole;
  final DateRange? dateRange;
  final String? description;

  TransactionFilterRequestDTO({
    this.householdId,
    this.userTransactionRole,
    this.dateRange,
    this.description,
  });

  factory TransactionFilterRequestDTO.fromJson(Map<String, dynamic> json) =>
      _$TransactionFilterRequestDTOFromJson(json);

  Map<String, dynamic> toJson() => _$TransactionFilterRequestDTOToJson(this);
}
